### PR TITLE
[SPARK-30744][SQL] Optimize AnalyzePartitionCommand by calculating location sizes in parallel

### PR DESCRIPTION
**Pull request by Ngone51**
### What changes were proposed in this pull request?

Use `CommandUtils.calculateTotalLocationSize` for `AnalyzePartitionCommand` in order to calculate location sizes in parallel.

### Why are the changes needed?

For better performance.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Pass Jenkins.

### CODE REVIEW

**Comment by Ngone51** [comment]
cc @cloud-fan @xuanyuanking

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala [hunk 1]
```diff
@@ -140,11 +140,11 @@ object CommandUtils extends Logging {
   def calculateTotalLocationSize(
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala]: the name should be `calculateLocationsSize`

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala [hunk 2]
```diff
@@ -153,11 +153,12 @@ object CommandUtils extends Logging {
    * for each path.
    * @param sparkSession the [[SparkSession]]
    * @param paths the Seq of [[Option[Path]]]s
-   * @return total size of all partitions
+   * @return a Seq of same size as `paths` where i-th element is total size of `paths(i)` or 0
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala]: same size?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala [hunk 3]
```diff
@@ -153,11 +153,12 @@ object CommandUtils extends Logging {
...
+   *         if `paths(i)` is None
    */
   def calculateLocationSizeParallel(
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala]: `calculateLocationsSizeParallel `

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala (hunk 2 above)
Comment (Ngone51) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala]: the returned `Seq`.size is equal to `paths`.size.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala (hunk 1 above)
Comment (Ngone51) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala]: rename to `calculateMultipleLocationSizes` after discuss with wenchen offline.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala [hunk 4]
```diff
@@ -153,11 +153,12 @@ object CommandUtils extends Logging {
    * for each path.
    * @param sparkSession the [[SparkSession]]
    * @param paths the Seq of [[Option[Path]]]s
-   * @return total size of all partitions
+   * @return a Seq of same size as `paths` where i-th element is total size of `paths(i)` or 0
+   *         if `paths(i)` is None
    */
-  def calculateLocationSizeParallel(
+  def calculateLocationsSizeInParallel(
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala]: let's be consistent `calculateMultipleLocationSizesInParallel`

**Comment by Ngone51** [comment]
Jenkins, retest this please.

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by Ngone51** [comment]
thanks!

**Pull request closed without merging**